modules/plugin/coverage-multidim/netcdf/src/main/java/org/geotools/imageio/netcdf/utilities/NetCDFUtilities.java

```diff
@@ -453,7 +489,12 @@ public static int getDimensionLength(Variable var, final int dimensionIndex) {
      */
     public static int getRawDataType(final VariableIF variable) {
         VariableDS ds = (VariableDS) variable;
-        final DataType type = ds.getOriginalDataType();
+        final DataType type;
+        if (Boolean.getBoolean(ENHANCE_SCALE_MISSING)) {
+            type = ds.getDataType();
```

Comment:
That's what I originally had, but unfortunately it's then not programmatically testable. I suppose I could make the variable not final? Neither is my favorite solution.